### ISSUE TITLE
please support the usage of `vim -t symbol`

### ISSUE THREAD

**Issue by chinyu0704** [ludovicchabant/vim-gutentags#331]
hi,
please support the usage of `vim -t symbol`, some people are used to it.
For now when I use this pattern, I get
 
![image](https://user-images.githubusercontent.com/19435140/179007640-5f4386dc-4afd-4ca5-9eaa-f1b4e4e10fca.png)

so I check the ctag database, and the new tag file generated by gutentags(not in current dir) was not there
![image](https://user-images.githubusercontent.com/19435140/179008091-cf546d88-c8e0-460b-b41a-aaf3dbc1a5b7.png)

**Comment by jdholtz** [comment]
I've found a workaround for this from [here](https://stackoverflow.com/a/40928964). You can add this to your `.vimrc`: `execute 'set tags+='.substitute(glob(g:gutentags_cache_dir .. '/*tags'), "\n", ',', 'g')`.

This would add every tags file ending with `tags` in your `gutentags_cache_dir` to the `tags` variable. Unfortunately, this would remove the feature of only adding the tags file for the current project, but this issue would have to be fixed within the script itself.

**Comment by chinyu0704** [comment]
this works, thanks @jdholtz

**Comment by ludovicchabant** [comment]
Right now gutentags tries to stay out of the way until it's needed, i.e. when you open a file that belongs to a project that needs tag support. Supporting the `-t` argument would require gutentags to always initialize and scan stuff on startup -- I don't see any obvious way to figure that `-t` was passed (surprisingly enough, `argc()` and `argv()` come out empty for that...). I'm not ready to refactor the plugin to work that way, but it's worth keeping in mind.

**Comment by jdholtz** [comment]
Hey @ludovicchabant. I’m not familiar with vimscript at all, but I was searching for a way to detect CLI arguments and [this](https://stackoverflow.com/questions/10357374/is-it-possible-to-access-vims-command-line-arguments-in-vimscript) might help. 

If you do somehow detect the `-t` flag, you could implement this by getting the CWD and recursively searching up to find a valid project directory to load the tags from.

**Comment by jdholtz** [comment]
You can find if `-t` is passed as a flag by doing `index (v:argv, ‘-t’) >= 0`. Although, this might only work with `-t {tag}` and not `-t{tag}`, I’m sure some pattern matching could solve this. 

However, when I tried implementing this, the tags variable gets set correctly, but it seems like vim searches for the tag before the plug-in has initialized, because the tag is still not found (even though doing :tag {tag} in the buffer does work).

Let me know if you have any solutions to this (unless this is what you mean by not wanting to initialize the tags unless it is necessary) and I can submit a PR for it.

**Comment by cassepipe** [comment]
I don't quite understand the problem. Why not use `tj` or `ts` or any other variant ?
I am myself a user of `vim -t` in the terminal but what has is got to do with gutentags ?

**Comment by jdholtz** [comment]
@cassepipe this problem occurs when the tags file is not in the current directory that `vim -t` is executed in (when using the`g:gutentags_cache_dir` option to store tags elsewhere besides the project root)

**Comment by cassepipe** [comment]
So the problem comes from `vim` not being able to find tags with the -t option when the tags is not
in the current directory, right ?
What about changing `g:gutentags_cache_dir` to the current directory ?

**Comment by jdholtz** [comment]
Yes, but many people want their tags stored in a certain directory, not the project root. Users shouldn’t be limited on how they can use this plugin if they want to use the `-t` flag with vim.

**Comment by jdholtz** [comment]
Hey @ludovicchabant, any updates after my comment [here](https://github.com/ludovicchabant/vim-gutentags/issues/331#issuecomment-1330036785) ?
